samples/mqtt/pkcs11_connect/main.cpp

```diff
@@ -199,99 +179,24 @@ int main(int argc, char *argv[])
     fprintf(stdout, "Connecting...\n");
     if (!connection->Connect(clientId.c_str(), false /*cleanSession*/, 1000 /*keepAliveTimeSecs*/))
     {
-        fprintf(stderr, "Failed to kick off MQTT async Connect(): %s\n", ErrorDebugString(connection->LastError()));
+        fprintf(stderr, "MQTT Connection failed with error %s\n", ErrorDebugString(connection->LastError()));
         exit(-1);
     }
 
-    // wait for the OnConnectionCompleted callback to fire, which sets connectionCompletedPromise...
-    if (connectionCompletedPromise.get_future().get() == false)
+    if (connectionCompletedPromise.get_future().get())
     {
-        fprintf(stderr, "Connection failed\n");
-        exit(-1);
-    }
```

Comment:
[style] go back to the old way of early-exit, instead of embedding the rest of the main program in an indented block.

early-exit is much easier to read
other people have other opinions, but they're wrong